### PR TITLE
swarm/network, swarm/storage: Preserve opentracing contexts

### PR DESCRIPTION
**Pull request by nolash**
This PR is an incremental part of https://github.com/ethersphere/go-ethereum/issues/1181 whose description is:

> At the moment if you send a GET request to Swarm for a specific content, you don't get proper tracing information in Jaeger, because the context propagation is broken.

> We need to fix this, so that we can see detailed traces of specific GET (or POST) requests and compare fast requests to slow requests in order to understand why we have a high variance for upload/download in smoke tests.

----

Specifically, the effect of this PR should be that the current context path is in correct hierarchical order:

1. HTTP GET
2. LazyChunkReader Read/Size/Seek
3. Stream request
4. Stream deliver

Furthermore, within the stream package it also makes sure the trace span finish executes _after_ protocol message transmission has been completed (before it only measured the time it took to insert a message into the priority queue)

### CODE REVIEW

File: swarm/network/stream/peer.go [hunk 1]
```diff
@@ -177,11 +186,6 @@ func (p *Peer) SendPriority(ctx context.Context, msg interface{}, priority uint8
 
 // SendOfferedHashes sends OfferedHashesMsg protocol msg
 func (p *Peer) SendOfferedHashes(s *server, f, t uint64) error {
-	var sp opentracing.Span
```
Comment (nonsense) [inline swarm/network/stream/peer.go]: Why are you removing this span? I know it is not very useful right now, as it is not to the other messages, but I don't think we should remove it.

Comment (nolash) [reply swarm/network/stream/peer.go]: I put it back

**Review by nonsense — APPROVED** [review]

**Comment by nonsense** [comment]
Great work! I ran the smoke tests on your branch and they are passing (meaning no regressions). This is a big improvement in terms of tracing.

screen shot 2019-02-08 at 14 56 50

**Review by skylenet — APPROVED** [review]

**Comment by nolash** [comment]
Submitted upstream https://github.com/ethereum/go-ethereum/pull/19022

**Pull request closed without merging**